docs/general/xs/beginner.md

```diff
@@ -1197,6 +1197,8 @@ For the name of the array, any string may be used but all names must be unique
 
 All of these functions return an integer which is the ID of the array that you just created. This ID is used to perform other operations on the array. You want to store this integer in a variable that is named apty so that you can use the array later.
 
+Arrays are global and never go out of scope (only their IDs do). As long as an array was created and you know its ID it can be accessed within any scope. So it is better to not create them in a code that will be executing multiple times as that can create memory leaks.
```

Comment:
I'd rewrite this in a way where stating "So be careful when creating arrays within rules, for loops and other iterative statements". Something like that. There are a lot of use cases for creating arrays within looping code without it being a "memory leak" (in the sense that array references get lost).